### PR TITLE
added spec for publish metrics

### PR DESCRIPTION
**Pull request by jenjenp**

### CODE REVIEW

File: services.yaml [hunk 1]
```diff
@@ -1368,6 +1368,71 @@ paths:
...
+          schema:
+            $ref: "#/definitions/PublishMetricsSettings"
+      responses:
+        200:
+          $ref: "#/responses/OK"
+        401:
+          $ref: "#/responses/UnauthorizedError"
+        404:
+          $ref: "#/responses/NotFoundError"
+  /services/{env_id}/settings/publish-metrics/available_formats:
```
Comment (chook) [inline services.yaml:1484]: does this really require an {env_id} ?

File: services.yaml [hunk 2]
```diff
@@ -3218,6 +3283,52 @@ definitions:
         items:
           type: string
           example: MyApp
+  PublishMetricsSettings:
+    type: object
+    properties:
+      enable:
+        type: boolean
+        example: true
+      server:
```
Comment (chook) [inline services.yaml]: it's actually `host:port` so i'm not sure "server" is a good term

File: services.yaml [hunk 3]
```diff
@@ -3218,6 +3283,52 @@ definitions:
...
+        type: string
+        example: 107.23.209.117.8125
```
Comment (chook) [inline services.yaml]: 107.23.209.117.8125 -> 107.23.209.117:8125

File: services.yaml [hunk 4]
```diff
@@ -3218,6 +3283,52 @@ definitions:
...
+      metrics:
+        $ref: "#/definitions/Metrics"
+  MetricFormats:
+    type: array
+    items:
+      type: object
+      properties:
+        output_format:
+          type: string
+          example: splunk
```
Comment (chook) [inline services.yaml:3616]: Splunk

File: services.yaml [hunk 5]
```diff
@@ -3218,6 +3283,52 @@ definitions:
...
+    type: array
+    items:
+      type: object
+      properties:
+        name:
+          type: string
+        pattern:
+          type:  string
+    example:
+      - name: views
```
Comment (chook) [inline services.yaml]: all the names should be capitalized correctly

File: services.yaml [hunk 6]
```diff
@@ -3218,6 +3283,52 @@ definitions:
...
+          $ref: "#/definitions/Metrics"
+  Metrics:
+    type: array
+    items:
+      type: object
+      properties:
+        name:
+          type: string
+        pattern:
+          type:  string
```
Comment (chook) [inline services.yaml]: double space

File: services.yaml [hunk 7]
```diff
@@ -1368,6 +1368,71 @@ paths:
...
+            $ref: "#/definitions/PublishMetricsSettings"
+      responses:
+        200:
+          $ref: "#/responses/OK"
+        401:
+          $ref: "#/responses/UnauthorizedError"
+        404:
+          $ref: "#/responses/NotFoundError"
+  /services/{env_id}/settings/publish-metrics/available_formats:
+    get:
```
Comment (chook) [inline services.yaml:1485]: what about a POST to actually set them? (by defaults, or custom edits)

**Review by chook — CHANGES_REQUESTED** [review]

File: services.yaml (hunk 1 above)
Comment (jenjenp) [reply services.yaml:1484]: Do you think it shall be under /global/... url? 
In app ui i see that the dialog is set differently when moving between keys, so isn't it a valid use case?

File: services.yaml (hunk 2 above)
Comment (jenjenp) [reply services.yaml]: updated + added description to this field

File: services.yaml (hunk 3 above)
Comment (jenjenp) [reply services.yaml]: fixed

File: services.yaml (hunk 4 above)
Comment (jenjenp) [reply services.yaml:3616]: fixed

File: services.yaml (hunk 5 above)
Comment (jenjenp) [reply services.yaml]: fixed

File: services.yaml (hunk 6 above)
Comment (jenjenp) [reply services.yaml]: fixed

File: services.yaml (hunk 7 above)
Comment (jenjenp) [reply services.yaml:1485]: Actually i thought about it, but i think that using the POST for /services/{env_id}/settings/publish-metrics will be enough, not sure we need additional call

File: services.yaml (hunk 1 above)
Comment (chook) [reply services.yaml:1484]: if so then fine

**Review by chook — APPROVED** [review]

File: services.yaml [hunk 8]
```diff
@@ -1437,6 +1437,71 @@ paths:
...
+        - in: path
+          name: env_id
+          description: Environment to use
+          type: string
+          required: true
+      responses:
+        200:
+          description: List of Available formats and metrics
+          schema:
+            $ref: "#/definitions/MetricFormats"
```
Comment (chook) [inline services.yaml]: i would rename this as PublishMetricFormats

File: services.yaml [hunk 9]
```diff
@@ -3517,6 +3582,54 @@ definitions:
             type: "string"
             example: "label=sample_label_name"
         type: "object"
+  PublishMetricsSettings:
+    type: object
+    properties:
+      enable:
```
Comment (chook) [inline services.yaml]: enabled?

File: services.yaml [hunk 10]
```diff
@@ -3517,6 +3582,54 @@ definitions:
...
+    type: array
+    items:
+      type: object
+      properties:
+        name:
+          type: string
+        pattern:
+          type: string
+    example:
+      - name: Views
```
Comment (chook) [inline services.yaml:3631]: what does name: Views mean?

**Review by chook — CHANGES_REQUESTED** [review]

File: services.yaml (hunk 8 above)
Comment (jenjenp) [reply services.yaml]: done

File: services.yaml (hunk 10 above)
Comment (jenjenp) [reply services.yaml:3631]: not sure i got you. That's an example of metric structure.

Comment (jenjenp) [reply services.yaml:3631]: Anyway, i added a description.

File: services.yaml [hunk 11]
```diff
@@ -3234,9 +3234,17 @@ definitions:
...
           - USER
         example: USER
-      color: #consider adding available colors as enum
+      color:
         type: string
         example: RED
+      created_on:
+        type: string
+        description: Label creation time. Date in ISO8601 format (yyyy-MM-ddTHH:mm:ss.SSSZ)
+        example: 2018-02-28T22:47:09.038+02:00
```
Comment (chook) [inline services.yaml]: Z

**Review by chook — APPROVED** [review]

**Pull request merged**